kit/src/components/embeds/link_embed/get_link_data.rs

```diff
@@ -0,0 +1,110 @@
+use std::error::Error;
+
+use reqwest::Url;
+use scraper::{Html, Selector};
+
+pub async fn get_image_data(document: Html, meta_selector: Selector) -> Option<String> {
+    let mut image = None;
+
+    for element in document.select(&meta_selector) {
+        let prop_attr_opt = element.value().attr("property");
+        let content_attr_opt = element.value().attr("content");
+
+        if let (Some(prop_attr), Some(content_attr)) = (prop_attr_opt, content_attr_opt) {
+            if prop_attr == "og:image" {
+                image = Some(content_attr.to_string());
+            }
+        }
+    }
+
+    image
+}
+
+pub async fn get_title_data(document: Html, meta_selector: Selector) -> String {
+    let title_selector = match Selector::parse("title") {
+        Ok(data) => data,
+        Err(_) => return String::new(),
+    };
+
+    let mut title = String::new();
+
+    for element in document.select(&meta_selector) {
+        let prop_attr_opt = element.value().attr("property");
+        let content_attr_opt = element.value().attr("content");
+
+        if let (Some(prop_attr), Some(content_attr)) = (prop_attr_opt, content_attr_opt) {
+            if prop_attr == "og:title" {
+                title = content_attr.to_string();
+            }
+        }
+    }
+
+    if title.is_empty() {
+        if let Some(element) = document.select(&title_selector).next() {
+            title = element.inner_html();
+        }
+    }
+
+    title
+}
+
+pub async fn get_description_data(document: Html, meta_selector: Selector) -> String {
```

Comment:
Ditto for not needing an async signature if nothing in the function is awaiting
```suggestion
pub fn get_description_data(document: Html, meta_selector: Selector) -> String {
```